platform/src/main/java/org/hillview/storage/ParquetFileLoader.java

```diff
@@ -153,19 +214,51 @@ private static void appendGroup(
     private static ColumnDescription getColumnDescription(ColumnDescriptor cd) {
         String name = String.join("", cd.getPath());  // this should contain a single String
         ContentsKind kind;
-        switch (cd.getPrimitiveType().getPrimitiveTypeName()) {
+        PrimitiveType primitiveType = cd.getPrimitiveType();
+        LogicalTypeAnnotation logicalType = primitiveType.getLogicalTypeAnnotation();
+        switch (primitiveType.getPrimitiveTypeName()) {
             case INT64:
+                if (logicalType instanceof TimestampLogicalTypeAnnotation) {
+                    if (((TimestampLogicalTypeAnnotation) logicalType).isAdjustedToUTC()) {
+                        kind = ContentsKind.Date;
+                    } else {
+                        kind = ContentsKind.LocalDate;
+                    }
+                } else if (logicalType instanceof TimeLogicalTypeAnnotation) {
+                    kind = ContentsKind.Time;
+                } else {
+                    kind = ContentsKind.Double;
+                }
+                break;
             case FLOAT:
             case DOUBLE:
                 kind = ContentsKind.Double;
                 break;
             case INT32:
-                kind = ContentsKind.Integer;
+                if (logicalType instanceof TimeLogicalTypeAnnotation) {
+                    kind = ContentsKind.Time;
+                } else {
+                    kind = ContentsKind.Integer;
+                }
                 break;
             case BOOLEAN:
-            case BINARY:
             case FIXED_LEN_BYTE_ARRAY:
-                kind = ContentsKind.String;
+                if (logicalType instanceof IntervalLogicalTypeAnnotation) {
+                    kind = ContentsKind.Duration;
+                } else {
+                    kind = ContentsKind.String;
+                }
+                break;
+            case BINARY:
+                if (logicalType instanceof StringLogicalTypeAnnotation) {
+                    kind = ContentsKind.String;
+                } else if (logicalType instanceof JsonLogicalTypeAnnotation) {
+                    kind = ContentsKind.Json;
+                } else {
+                    HillviewLogger.instance.warn("Encountered BINARY field with no or unhandled logical type annotation: attempting to parse as string",
```

Comment:
this will also give lots of warnings

Reply:
Actually this function is the conversion from parquet schema to hillview schema. It only happens once and will log at most one warning message for each column